### PR TITLE
Handle null value min/max statistics in Delta Lake tables

### PR DESCRIPTION
**Pull request by vkorukanti**
## Description
Not able to read a Delta table containing min/max values as `nulls` in statistics written in the log. This is a special case where the data file contains a single row and one or more columns are `null` in that row. In this case `nullCount` is set correctly, but some engines (in this case Apache Spark™) write `minValue` and `maxValue` as null depending upon the JSON writer configuration (in this case `spark.sql.jsonGenerator.ignoreNullFields=false`).

Trino uses the Guava's `ImmutableMap` for post-processing of the statistics from the Delta log which doesn't accept `null` values. The fix is to filter out the null statistics for `minValues` and `maxValues`. This is like as if the `min` and `max` statistics are missing for the column.

Added unit and integration tests.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java [hunk 1]
```diff
@@ -479,6 +479,7 @@ public static <T> Map<CanonicalColumnName, T> toCanonicalNameKeyedMap(Map<String
     public static <T> Map<CanonicalColumnName, T> toCanonicalNameKeyedMap(Map<String, T> map, Map<String, CanonicalColumnName> canonicalColumnNames)
     {
         return map.entrySet().stream()
+                .filter(entry -> entry.getValue() != null)
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java:482]: Can we remove another `toCanonicalNameKeyedMap(Map<String, T> map)` in a preparatory commit?

File: plugin/trino-delta-lake/src/test/resources/deltalake/stats_with_minmax_nulls/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+```
+bin/spark-sql --packages io.delta:delta-core_2.12:2.4.0 \
+  --conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" \
+  --conf "spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog" \
+  --conf "spark.sql.jsonGenerator.ignoreNullFields=false"
+```
+
+```sql
+CREATE TABLE delta.`/tmp/stats-with-minmax-nulls` USING DELTA AS SELECT col1 as id, col2 as id2 FROM VALUES (0, 1),(1,2),(3, 4);
+INSERT INTO delta.`/tmp/stats-with-minmax-nulls` select null, null;
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/resources/deltalake/stats_with_minmax_nulls/README.md:12]: optional: We could add another test reading a checkpoint parquet file having null value in min/max statistics.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 3]
```diff
@@ -709,6 +709,29 @@ private void testCorruptedTableLocation(String tableName, Path tableLocation, bo
...
+                format("CALL system.register_table('%s', '%s', '%s')", getSession().getSchema().orElseThrow(), tableName, tableLocation));
+
+        assertThat(query("SELECT * FROM " + tableName))
+        .matches("""
+                    VALUES
+                    (0, 1),
+                    (1, 2),
+                    (3, 4),
+                    (NULL, NULL)
+                    """);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:732]: Can we verify `SHOW STATS FOR` statement as well? I believe this statement also failed without this PR.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 4]
```diff
@@ -709,6 +709,29 @@ private void testCorruptedTableLocation(String tableName, Path tableLocation, bo
         }
     }
 
+    /**
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:712]: Please squash commits into one.

**Comment by vkorukanti** [comment]
@ebyhr Thanks for reviewing! Addressed the comments, PTAL.

File: plugin/trino-delta-lake/src/test/resources/deltalake/stats_with_minmax_nulls/README.md (hunk 2 above)
Comment (vkorukanti) [reply plugin/trino-delta-lake/src/test/resources/deltalake/stats_with_minmax_nulls/README.md]: Added a test table with checkpoint. Currently, it just has stats as JSON string in the checkpoint. Delta OSS doesn't have the feature to write the parsed stats in checkpoint. Will followup with a test table with parsed stats in the checkpoint file in the next PR.

**Pull request merged**